test/test_jit_cuda_fuser.py

```diff
@@ -935,48 +935,39 @@ def test_binary_ops(self):
     @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
                      "Requires fusion optimization pass to be effective")
     def test_binary_bitwise(self):
-        def jit_or(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
-            return (x & y) | z
-
-        def jit_xor(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
-            return (x & y) ^ z
-
-        def jit_lshift(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
-            return (x & y) << z
-
-        def jit_rshift(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
-            return (x & y) >> z
-
-        for jit_func in [jit_or, jit_xor, jit_lshift, jit_rshift]:
-            x = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(5).to(torch.long)
-            y = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(5).to(torch.long)
-            z = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(2).to(torch.long)
-
-            jitted = torch.jit.script(jit_func)
-            jit_o = jitted(x, y, z)
-            jit_o = jitted(x, y, z)
-            o = jit_func(x, y, z)
-            self.assertEqual(o, jit_o)
-            self.assertGraphContains(jitted.graph_for(x, y, z), FUSION_GUARD)
-
-        # We shouldn't need this redefinition of the function, but otherwise it won't recompile for a new type
```

Comment:
Why do we have to redefine function to get a recompile? Shouldn't nvFuser always recompile when dtype changes? This sounds like a bug to me.

Reply:
What do you mean by redefine?  Do you mean apply `torch.jit.script(func)`?

Reply:
I think it means rerun the `def function_name(....` code. I don't know why, but I have a feeling that there is some mechanism somewhere in TorchScript or nvFuser associating a Python's function object with its input tensor dtypes, and once this func obj -> dtype mapping is established, it will not change.

Reply:
It's not nvfuser, but torchscript.

And the problem is mostly just about testing. For updated dtype will go through TorchScript with Guard-false-block (the gracefull fallback). So we still should be running the same thing. But none of this is visible directly from `graph_for`, since they don't show optimized fallback.
Our test code doesn't actually look at the kernel we executed tho, so if we look at the TS graph, it's easier to rely on the recompilation.

Reply:
But similarly, you can try to use the disable cache thing to make it cleaner.

Reply:
OK, gotcha! Thanks @jjsjann123